editor/src/components/canvas/canvas-strategies/strategies/group-conversion-helpers.ts

```diff
@@ -607,24 +632,237 @@ export function groupConversionCommands(
+          width: newLocalRectangleForGroup.width,
+          height: newLocalRectangleForGroup.height,
+        },
+        emptyComments,
+      ),
+    }),
+    childComponents.map((c) => c.element),
+  )
+
+  // if any group child was a child of the group's target parent, let's use the child's original index for the insertion
+  const anyChildIndexInTargetParent: Absolute | undefined = mapDropNulls((child) => {
+    const childIsTheChildOfTargetParent = EP.isParentOf(
+      parentPath.intendedParentPath,
+      child.metadata.elementPath,
+    )
+    if (!childIsTheChildOfTargetParent) {
+      return null
+    }
+
+    const indexInParent = MetadataUtils.getIndexInParent(
+      metadata,
+      elementPathTrees,
+      child.metadata.elementPath,
+    )
+    if (indexInParent < 0) {
+      return null
+    }
+    return absolute(indexInParent)
+  }, childComponents).at(0)
+
+  const indexPosition = anyChildIndexInTargetParent ?? back()
+
+  // insert a group in the common ancestor
+  const insertGroupCommand = addElement('always', parentPath, group, {
+    importsToAdd: GroupImport,
+    indexPosition: indexPosition,
   })
+
+  const groupPath = EP.appendToPath(parentPath.intendedParentPath, group.uid) // TODO does this work if the parentPath is a ConditionalClauseInsertionPath?
+
+  const pinChangeCommands = childComponents.flatMap((childComponent) => {
+    const newChildPath = EP.appendToPath(groupPath, childComponent.element.uid)
+    const childLocalRect: LocalRectangle = canvasRectangleToLocalRectangle(
+      forceFiniteRectangle(childComponent.metadata.globalFrame),
+      globalBoundingBoxOfAllElementsToBeWrapped,
+    )
+    return [
+      // make the child `position: absolute`
+      setProperty('always', newChildPath, PP.create('style', 'position'), 'absolute'),
+      // set child pins to match their intended new local rectangle
+      ...setElementPinsForLocalRectangleEnsureTwoPinsPerDimension(
+        newChildPath,
+        childComponent.element.props,
+        childLocalRect,
+        sizeFromRectangle(newLocalRectangleForGroup),
+        null,
+      ),
+    ]
+  })
+
+  return applyCommandsAction([...deleteCommands, insertGroupCommand, ...pinChangeCommands])
+}
+
+function setElementPinsForLocalRectangleEnsureTwoPinsPerDimension(
+  target: ElementPath,
+  elementCurrentProps: JSXAttributes,
+  localFrame: LocalRectangle,
+  parentSize: Size,
+  parentFlexDirection: FlexDirection | null,
+): Array<CanvasCommand> {
+  // ensure at least two pins per dimension
+  const mustHavePins = ensureAtLeastTwoPinsForEdgePosition(
+    right(elementCurrentProps),
+    EdgePositionBottomRight,
+  )
+
+  // TODO retarget Fragments
+  const result = [
+    setCssLengthProperty(
+      'always',
+      target,
+      PP.create('style', 'left'),
+      setValueKeepingOriginalUnit(localFrame.x, parentSize.width),
+      parentFlexDirection,
+      mustHavePins.includes('left') ? 'create-if-not-existing' : 'do-not-create-if-doesnt-exist',
```

Comment:
my suggestion would be to make a small generator function for these, e.g.
```ts
function setPinCssProperty(pin: 'top' | 'left' | 'right' | ' bottom', x: number, y: number)
```
and return four of those directly

Reply:
fixed!